app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -7,7 +7,7 @@ module TwoFactorAuthenticatable
     before_action :require_current_password, if: :current_password_required?
     before_action :check_already_authenticated
     before_action :reset_attempt_count_if_user_no_longer_locked_out, only: :create
-    before_action :apply_secure_headers_override, only: :show
+    before_action :apply_secure_headers_override, only: [:show, :create]
```

Comment:
If this fix is in TwoFactorAuthenticatable, wouldn't it also affect SAML requests? I am not able to reproduce this coming from sp.int.login.gov.

I think what we want to do here is to modify the `before_action` in `OpenidConnect::AuthorizationController` which defines its own `apply_secure_headers_override` method.